pandas/tests/io/test_fsspec.py

```diff
@@ -72,7 +72,9 @@ def test_read_csv(cleared_fs, df1):
         w.write(text)
     df2 = read_csv("memory://test/test.csv", parse_dates=["dt"])
 
-    tm.assert_frame_equal(df1, df2)
+    expected = df1.copy()
```

Comment:
I wonder if this will be problematic that this doesn't round-trip the same way any more.